elfi/methods/results.py

```diff
@@ -203,6 +204,36 @@ def sample_means_array(self):
         """
         return np.array(list(self.sample_means.values()))
 
+    def save_samples(self, name=None, type_='csv'):
+        """Save samples in csv or json file formats.
+
+        Parameters
+        ----------
+        name : file name to be saved
+        type_ : file format to be saved
+
+        """
+        import json
+        import csv
+
+        if not name:
+            name = self.__class__.__name__.lower() + '_' + 'samples.' + type_
+
+        if type_ == 'csv':
+            with open(name, 'w', newline='') as f:
+                w = csv.writer(f)
+                w.writerow(self.samples.keys())
+                w.writerows(itertools.zip_longest(*self.samples.values(), fillvalue=''))
```

Comment:
`np.savetxt` would give more freedom, e.g. something else than comma as the delimiter, user-defined precision etc.. 

But this is also fine.